lib/gtfs/data.ex

```diff
@@ -0,0 +1,266 @@
+defmodule Gtfs.Data do
+  @moduledoc """
+  GTFS Data management
+  """
+  require Logger
+
+  alias Gtfs.CacheFile
+  alias Gtfs.Csv
+  alias Gtfs.Direction
+  alias Gtfs.Helpers
+  alias Gtfs.Route
+  alias Gtfs.RoutePattern
+  alias Gtfs.Stop
+  alias Gtfs.StopTime
+  alias Gtfs.Timepoint
+  alias Gtfs.Trip
+
+  @type t :: %__MODULE__{
+          routes: [Route.t()],
+          route_patterns: [RoutePattern.t()],
+          stops: [Stop.t()],
+          trip_stops: trip_stops(),
```

Comment:
We separate the `stop_times` into `trip_stops` and `trip_timepoints`, just to combine them again when calculating the next timepoint.

We could instead leave them combined in one `trip_stop_times: %{Trip.id() => [StopTime.t()]}` field, where `@type StopTime.t :: %{stop_id: Stop.id(), timepoint_id: Timepoint.id()}`

The existing code that makes ladders would then filter the stop_times to just the ones with timepoints.